server/php-runner.cpp

```diff
@@ -522,10 +522,14 @@ void write_str(int fd, const char *s) noexcept {
   write(fd, s, std::min(strlen(s), size_t{1000}));
 }
 
+std::array<void *, 128> sigalrm_last_backtrace;
+int sigalrm_last_backtrace_size;
+
 namespace kphp_runtime_signal_handlers {
 
 static void sigalrm_handler(int signum) {
   kwrite_str(2, "in sigalrm_handler\n");
+  sigalrm_last_backtrace_size = fast_backtrace(sigalrm_last_backtrace.data(), sigalrm_last_backtrace.size());
```

Comment:
are you sure that it's [async signal safe](https://man7.org/linux/man-pages/man7/signal-safety.7.html)?